pymc/model.py

```diff
@@ -749,11 +749,10 @@ def logpt(
                         f"Requested variable {var} not found among the model variables"
                     )
 
-        rv_logps = []
+        rv_logps: List[TensorVariable] = []
         if rv_values:
             rv_logps = joint_logpt(list(rv_values.keys()), rv_values, sum=False, jacobian=jacobian)
-            if not isinstance(rv_logps, list):
-                rv_logps = [rv_logps]
+            assert isinstance(rv_logps, list)
```

Comment:
I think this change is wrong. `joint_logpt` returns a single variable when there is only one rv

Reply:
Oh. I thought it was safe to assume that when `sum=False` it returns a list.
Should I make it do that?
-----------
Functions with dynamic return types are making it really hard to get types right.
Please do not add any more of them.

Reply:
Trust me I didn't want to add it

Reply:
Do we still need that assert or can mypy figure it out now?

Reply:
Mypy can not figure it out without either an assert, isinstance check or cast.

As I said. Functions with dynamic types are a typing nightmare.

Reply:
I am just uneasy with an assert. Have you seen any discussion about cast vs asserts for typing?

Reply:
At runtime the cast becomes a dummy function call not doing anything.
The assert takes a tiny bit longer, but with the added benefit of raising an error when the type is not actually as expected.
For that reason, and because this code is not time critical at the microsecond level, I prefer the asserts.
Note that the assert I added detected my incorrect assumption, giving me the same information as your initial comment.